src/components/DropdownFilterContainer.ts

```diff
@@ -0,0 +1,147 @@
+
+export interface ListView extends mxui.widget._WidgetBase {
+    datasource: {
+        xpathConstraints: string;
+    };
+    _datasource: {
+        _constraints: HybridConstraint | string;
+        _entity: string;
+        _setSize: number;
+        atEnd: () => boolean;
+        _pageSize: number;
+    };
+    _loadMore: () => void;
+    _onLoad: () => void;
+    _renderData: () => void;
+    update: () => void;
+}
+
+interface DropdownFilterState {
+    findingWidget: boolean;
+    targetGrid?: ListView;
+    targetNode?: HTMLElement;
+    validationPassed?: boolean;
+}
+
+export default class DropdownFilterContainer extends Component<DropdownFilterContainerProps, DropdownFilterState> {
+
+    constructor(props: DropdownFilterContainerProps) {
+        super(props);
+
+        this.state = { findingWidget: true };
+        this.handleChange = this.handleChange.bind(this);
+        dojoConnect.connect(props.mxform, "onNavigation", this, dojoLang.hitch(this, this.initDropdownFilter));
+    }
+
+    render() {
+        return createElement("div",
+            {
+                className: classNames("widget-dropdown-filter")
+            },
+            createElement(ValidateConfigs, {
+                ...this.props as DropdownFilterContainerProps,
+                filterNode: this.state.targetNode,
+                targetGrid: this.state.targetGrid,
+                targetGridName: this.props.targetGridName,
+                validate: !this.state.findingWidget,
+                values: this.props.values
+            }),
+            this.renderDropdownFilter()
+        );
+    }
+
+    private handleChange(value: string, attribute: string, comparison: comparisonOptions, constraint: string, filterMethod: string) {
+        if (comparison === "static") {
+            this.updateByConstraint(value, attribute, filterMethod);
+        } else {
+            this.updateByXpath(constraint);
+        }
+    }
+
+    private updateByXpath(constraint: string) {
+        this.state.targetGrid.datasource.xpathConstraints = constraint;
+    }
+
+    private updateByConstraint(value: string, attribute: string, filterMethod: string) {
+        const constraints: HybridConstraint = [];
+        if (this.state.targetGrid && this.state.targetGrid._datasource) {
+            const datasource = this.state.targetGrid._datasource;
+            if (window.device) {
+                constraints.push({
+                    attribute,
+                    operator: filterMethod,
+                    path: this.props.entity,
+                    value
+                });
+                datasource._constraints = value ? constraints : [];
+            } else {
+                let constraint: HybridConstraint | string = [];
+                constraint = `${filterMethod}(${attribute},'${value}')`;
+                datasource._constraints = value ? "[" + constraint + "]" : "";
```

Comment:
Simply overwriting the constraint is not supporting multiple filters on one list view